language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -489,10 +489,22 @@ class DottyLanguageServer extends LanguageServer
 
     val uriTrees = driver.openedTrees(uri)
 
-    val defs = Interactive.namedTrees(uriTrees, Include.empty)
+    // Excludes locals from synthetic symbols (with the exception of enums members)
+    // and erroroneous trees.
+    val excludeLocalsFromSyntheticSymbols = (n: NameTree) => {
+      val owner = n.symbol.owner
+      n.symbol.is(Case) || {
+        n.name != StdNames.nme.ERROR &&
```

Comment:
I don't think we ever want to return these ERROR trees, so we could add this check to the ones we do by default in https://github.com/lampepfl/dotty/blob/3a7dfd2c14e1bc146f79faef8fcad42efadcdaeb/compiler/src/dotty/tools/dotc/interactive/Interactive.scala#L187

Reply:
I'm going to move the condition. I wasn't sure this was the best way to check for an erroneous tree. Can you confirm this? Does this change impact tests other than the language server / worksheets?

Reply:
> I wasn't sure this was the best way to check for an erroneous tree. Can you confirm this?

It's the best way to deal with the error trees produced by Parser currently, typed error trees should already be handled.

> Does this change impact tests other than the language server / worksheets?

I guess we'll have to run the tests and see :).